### PR TITLE
Use `hasfield` to check whether `logger.stream` is legal

### PR DESCRIPTION
**Pull request by fonsp**
An `AbstractLogger` is not guaranteed to have this field defined.

This fixes an issue where setting a custom logger as global logger will break Suppressor.jl

### CODE REVIEW

**Comment by fonsp** [comment]
🥞

**Comment by Krastanov** [comment]
What is the status of this pull request? I am encountering this error when trying to use LoggingExtras https://github.com/oxinabox/LoggingExtras.jl/issues/47

**Comment by Krastanov** [comment]
@omus Sorry to bother you with this, but you are the only committer I see is part of JuliaIO. Any chance you can given an ETA on this pull request (from last October) or maybe directing me to the appropriate developer whom I should ping?

**Comment by tfiers** [comment]
For others discovering this PR: until it is merged, you can install @fonsp's patched version like so:
-  `pkg> add https://github.com/fonsp/Suppressor.jl.git#patch-1`,
-  or programmatically (_e.g._ in `atreplinit` in `startup.jl`):
   `Pkg.add(url="https://github.com/fonsp/Suppressor.jl.git", rev="patch-1")`.

**Comment by omus** [comment]
@Krastanov I apologize for missing your ping. I'll also make sure I watch this package as I didn't realize I was one of the only maintainers :S

**Comment by Krastanov** [comment]
Thanks, @omus ! Let me know if I can help (obviously maintainer status requires some vetting, but I can try to help, especially given how "pretty much complete" this project is)

File: src/Suppressor.jl
Comment (omus) [inline src/Suppressor.jl:28]: Just for reference normally I'd either use Compat.jl so you can use `hasproperty` or do:
```julia
if !isdefined(Base, :hasproperty)
    hasproperty(x, s::Symbol) = s in propertynames(x)
end
```
so you can use the nicer syntax. I'll forgo making change though as people have waited long enough for this.

**Review by omus — APPROVED** [review]

**Comment by omus** [comment]
Just noticed this package is still using Travis for CI. I'll merge this but address the CI situation before making a new release

**Comment by omus** [comment]
Fix has been registered in version 0.2.1

**Pull request merged**
